File: java_projects/glue_code/commons-validator/src/main/java/org/apache/commons/validator/routines/BigDecimalValidator.java
package org.apache.commons.validator.routines;

import java.math.BigDecimal;
import java.text.Format;
import java.util.Locale;
import org.apache.commons.validator.ContextInitializer;
import org.graalvm.polyglot.Value;

public class BigDecimalValidator extends AbstractNumberValidator {
  private static final BigDecimalValidator VALIDATOR = BigDecimalValidator.BigDecimalValidator2();
  private static final long serialVersionUID = -670320911490506772L;
  private static Value clz =
      ContextInitializer.getPythonClass("/routines/BigDecimalValidator.py", "BigDecimalValidator");
  private Value obj;

  public BigDecimalValidator(Value obj) {
    this.obj = obj;
  }

  public Value getPythonObject() {
    return obj;
  }

  protected Object processParsedValue(Object value, Format formatter) {
    //
    // BigDecimal decimal = null;
    // if (value instanceof Long) {
    // decimal = BigDecimal.valueOf(((Long) value).longValue());
    // } else {
    // decimal = new BigDecimal(value.toString());
    // }
    //
    // int scale = determineScale((NumberFormat) formatter);
    // if (scale >= 0) {
    // decimal = decimal.setScale(scale, BigDecimal.ROUND_DOWN);
    // }
    //
    // return decimal;
    //

    // TODO: Check the type mapping below!
    return obj.invokeMember("processParsedValue", value, formatter).as(Object.class);
  }

  public boolean maxValue(BigDecimal value, double max) {
    //
    // return (value.doubleValue() <= max);
    //

    // TODO: Check the type mapping below!
    return obj.invokeMember("maxValue", value, max).as(boolean.class);
  }

  public boolean minValue(BigDecimal value, double min) {
    //
    // return (value.doubleValue() >= min);
    //

    // TODO: Check the type mapping below!
    return obj.invokeMember("minValue", value, min).as(boolean.class);
  }

  public boolean isInRange(BigDecimal value, double min, double max) {
    //
    // return (value.doubleValue() >= min && value.doubleValue() <= max);
    //

    // TODO: Check the type mapping below!
    return obj.invokeMember("isInRange", value, min, max).as(boolean.class);
  }

  public BigDecimal validate3(String value, String pattern, Locale locale) {
    //
    // return (BigDecimal) parse(value, pattern, locale);
    //

    // TODO: Check the type mapping below!
    return obj.invokeMember("validate3", value, pattern, locale).as(BigDecimal.class);
  }

  public BigDecimal validate2(String value, Locale locale) {
    //
    // return (BigDecimal) parse(value, (String) null, locale);
    //

    // TODO: Check the type mapping below!
    return obj.invokeMember("validate2", value, locale).as(BigDecimal.class);
  }

  public BigDecimal validate1(String value, String pattern) {
    //
    // return (BigDecimal) parse(value, pattern, (Locale) null);
    //

    // TODO: Check the type mapping below!
    return obj.invokeMember("validate1", value, pattern).as(BigDecimal.class);
  }

  public BigDecimal validate0(String value) {
    //
    // return (BigDecimal) parse(value, (String) null, (Locale) null);
    //

    // TODO: Check the type mapping below!
    return obj.invokeMember("validate0", value).as(BigDecimal.class);
  }

  public static BigDecimalValidator BigDecimalValidator2() {
    //
    // return BigDecimalValidator1(true);
    //

    // TODO: Check the type mapping below!
    return clz.invokeMember("BigDecimalValidator2").as(BigDecimalValidator.class);
  }

  public static BigDecimalValidator BigDecimalValidator1(boolean strict) {
    //
    // return new BigDecimalValidator(strict, STANDARD_FORMAT, true);
    //

    // TODO: Check the type mapping below!
    return clz.invokeMember("BigDecimalValidator1", strict).as(BigDecimalValidator.class);
  }

  protected BigDecimalValidator(boolean strict, int formatType, boolean allowFractions) {
    //
    // super(strict, formatType, allowFractions);
    //

    this.obj = clz.invokeMember("__init__", strict, formatType, allowFractions);
  }

  public static BigDecimalValidator getInstance() {
    //
    // return VALIDATOR;
    //

    // TODO: Check the type mapping below!
    return clz.invokeMember("getInstance").as(BigDecimalValidator.class);
  }
}

File: data/recomposed_projects/deepseek-coder-33b-instruct/signature/commons-validator/src/main/org/apache/commons/validator/routines/BigDecimalValidator.py
from __future__ import annotations
import time
import locale
import re
import io
import numbers
import typing
from typing import *
import decimal
import datetime
from src.main.org.apache.commons.validator.routines.AbstractNumberValidator import *


class BigDecimalValidator(AbstractNumberValidator):

    __VALIDATOR: BigDecimalValidator = None
    __serialVersionUID: int = -670320911490506772

    @staticmethod
    def initialize_fields() -> None:
        BigDecimalValidator.__VALIDATOR: BigDecimalValidator = (
            BigDecimalValidator.BigDecimalValidator2()
        )

    def _processParsedValue(
        self, value: typing.Any, formatter: typing.Union[str, datetime.datetime]
    ) -> typing.Any:

        decimal_val = None
        if isinstance(value, int):
            decimal_val = decimal.Decimal(value)
        else:
            decimal_val = decimal.Decimal(str(value))

        scale = self._determineScale(formatter)
        if scale >= 0:
            decimal_val = decimal_val.quantize(
                decimal.Decimal(10) ** -scale, decimal.ROUND_DOWN
            )

        return decimal_val

    def maxValue(self, value: decimal.Decimal, max: float) -> bool:
        return value.to_eng_string() <= str(max)

    def minValue(self, value: decimal.Decimal, min: float) -> bool:
        return value.to_eng_string() >= str(min)

    def isInRange(self, value: decimal.Decimal, min: float, max: float) -> bool:
        return min <= value.doubleValue() <= max

    def validate3(
        self, value: str, pattern: str, locale: typing.Any
    ) -> decimal.Decimal:
        return self._parse(value, pattern, locale)

    def validate2(self, value: str, locale: typing.Any) -> decimal.Decimal:
        return self._parse(value, None, locale)

    def validate1(self, value: str, pattern: str) -> decimal.Decimal:
        return self._parse(value, pattern, None)

    def validate0(self, value: str) -> decimal.Decimal:
        return self._parse(value, None, None)

    @staticmethod
    def BigDecimalValidator2() -> BigDecimalValidator:
        return BigDecimalValidator.BigDecimalValidator1(True)

    @staticmethod
    def BigDecimalValidator1(strict: bool) -> BigDecimalValidator:
        return BigDecimalValidator(strict, BigDecimalValidator.STANDARD_FORMAT, True)

    def __init__(self, strict: bool, formatType: int, allowFractions: bool) -> None:
        super().__init__(strict, formatType, allowFractions)

    @staticmethod
    def getInstance() -> BigDecimalValidator:
        return BigDecimalValidator.__VALIDATOR


BigDecimalValidator.initialize_fields()
